include/libssh2.h

```diff
@@ -333,6 +333,22 @@ typedef struct _LIBSSH2_SK_SIG_INFO {
     void name(LIBSSH2_SESSION *session, LIBSSH2_CHANNEL *channel, \
               const char *shost, int sport, void **abstract)
 
+#define LIBSSH2_AUTHAGENT_FUNC(name) \
+   void name(LIBSSH2_SESSION *session, LIBSSH2_CHANNEL *channel, \
+             void **abstract)
+
+#define LIBSSH2_ADD_IDENTITIES_FUNC(name) \
+   void name(LIBSSH2_SESSION *session, LIBSSH2_CHANNEL *channel, \
+             void **abstract)
+
+#define LIBSSH2_AUTHAGENT_SIGN_FUNC(name) \
+   int name(LIBSSH2_SESSION* session, \
+            unsigned char *blob, unsigned int blen, \
+            const unsigned char *data, unsigned int dlen, \
```

Comment:
@MichaelBuckley I've added `const` here. Do you think this is okay?

Reply:
Yes, good catch. However, we will also need to add const to `libssh2_agent_sign()` to match. The `agent_sign()` function already marks the data parameter `const`.

Reply:
Yep, added that one already. Though we still need to sync the man page with this.